using-mobile-extensions/adobe-edge-consent/edge-consent-api-reference.md

```diff
@@ -0,0 +1,290 @@
+
+{% tab title="iOS" %}
+
+### Swift
+
+Use the MobileCore API to register the Edge Consent extension.
+
+#### Syntax
+
+```swift
+public static func registerExtensions(_ extensions: [NSObject.Type], _ completion: (() -> Void)? = nil)
+```
+
+#### Example
+
+```swift
+MobileCore.registerExtensions([Consent.self, ...], {
+  // processing after registration
+})
+```
+
+### Objective-C
+
+Use the AEPMobileCore API to register the Edge Consent extension.
+
+#### Syntax
+
+```swift
+public static func registerExtensions(_ extensions: [NSObject.Type], _ completion: (() -> Void)? = nil)
+```
+
+#### Example
+
+```objective-c
+[AEPMobileCore registerExtensions:@[AEPMobileEdgeConsent.class, ...] completion:^{
+  // processing after registration
+}];
+```
+
+{% endtab %}
+{% endtabs %}
+
+## getConsents<a id="getconsents"></a>
+
+Retrieves the current consent preferences stored in the Consent extension.
+
+{% tabs %}
+{% tab title="Android" %}
+### Java
+
+#### Syntax
+
+```java
+public static void getConsents(final AdobeCallback<Map<String, Object>> callback);
+```
+
+* _callback_ - callback invoked with the current consents of the extension. If an `AdobeCallbackWithError` is provided, an `AdobeError`, can be retruned in the eventuality of any error that occured while getting the user consents. The callback may be invoked on a different thread.
+
+#### Example
+
+```java
+Consent.getConsents(new AdobeCallback<Map<String, Object>>() {
+	@Override
+	public void call(Map<String, Object> currentConsents) {
+		// handle currentConsents
+	}
+});
+```
+{% endtab %}
+
+{% tab title="iOS" %}
+### Swift
+
+#### Syntax
+
+```swift
+static func getConsents(completion: @escaping ([String: Any]?, Error?) -> Void)
+```
+
+* _completion_ - Invoked with the current consent preferences or an `AEPError` if an unexpected error occurs or the request timed out. It may be invoked on a different thread.
+
+#### Example
+
+```swift
+Consent.getConsents { currentConsents, error in
+	// handle currentConsents
+}
+```
+
+### Objective-C
+
+#### Syntax
+
+```objective-c
+static func getConsents(completion: @escaping ([String: Any]?, Error?) -> Void)
+```
+
+#### Example
+
+```objective-c
+[AEPMobileEdgeConsent getConsents:^(NSDictionary *currentConsents, NSError *error){
+	// handle currentConsents
+}];
+```
+{% endtab %}
+{% endtabs %}
+
+## updateConsents<a id="updateConsents"></a>
+
+Merges the existing consents with the given consents. Duplicate keys will take the value of those passed in the API.
+
+> After a user has selected collect consent no (n), the SDK will not allow you to set the users collect consent to yes (y).
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+#### Syntax
+
+```java
+public static void update(final Map<String, Object> xdmFormattedConsents);
+```
+
+* *xdmFormattedConsents* - A `Map` of consents in predefined [Profile Consents XDM schema](https://github.com/adobe/xdm/blob/master/docs/reference/mixins/profile/profile-consents.schema.md).
```

Comment:
Minor: the `xdmFormattedConsents` parameter got renamed to `consents`
See https://github.com/adobe/aepsdk-edgeconsent-android/blob/dev/code/edgeconsent/src/main/java/com/adobe/marketing/mobile/edge/consent/Consent.java#L53-L59
Please update all occurrences